### ISSUE TITLE
Migrating from 6.1 to 7.2 does not remove/move reflect metadata polyfill

### ISSUE THREAD

**Issue by Astrophizz** [angular/angular-cli#13491]
# 🐞 Bug report

### Command (mark with an `x`)

```
- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [x] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?

No.

### Description

According to the [blog post](https://blog.angular.io/version-7-of-angular-cli-prompts-virtual-scroll-drag-and-drop-and-more-c594e22e7b8c) about Angular 7, updating to Angular 7 should remove the reflect metadata polyfill. The work in #12314, #12611, and #12625 suggests it might also get moved the the dev environment, though discussions around the code seem to disagree. No changes are made to polyfill.ts when updating.

## 🔬 Minimal Reproduction

Run `ng update --all --force` on an Angular 6 application (this works around other outstanding issues with `ng update`).

## 🌍 Your Environment

<pre><code>
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 6.2.5
Node: 10.15.0
OS: win32 x64
Angular: 6.1.10
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.6.8
@angular-devkit/build-angular     0.6.8
@angular-devkit/build-optimizer   0.6.8
@angular-devkit/core              0.6.8
@angular-devkit/schematics        0.6.8
@angular/cli                      6.2.5
@ngtools/webpack                  6.0.8
@schematics/angular               0.8.5
@schematics/update                0.8.5
rxjs                              6.3.3
typescript                        2.9.2
webpack                           4.8.3
</code></pre>

**Anything else relevant?**

Not that I can think of. It might be nice for the [Angular Update Guide](https://update.angular.io/) to explain how to manually migrate the polyfill if the automated migration does not do anything.

**Comment by alan-agius4** [comment]
Hi, I tried to replicate this and the `import 'core-js/es7/reflect';` was removed.

Can you setup a minimal repro please?

This might be related to your directory structure so its really important to get an accurate repro to diagnose this.

Thanks

**Comment by Astrophizz** [comment]
In attempting to create a minimal repro by amputating my application until it resembled a fresh Angular project and the migration still failing to remove the polyfill, I think I've narrowed in on the difference between a successful migration and one that doesn't remove the reflect polyfill. The key difference was whether I ran `ng update --all` without a pre-existing `.\node_modules` (**success**) or running `npm i` to populate `.\node_modules` before `ng update --all` (**failure**).

On a side note, I do find it odd that the migration removes the polyfill import but not the comment describing it.

**Comment by Astrophizz** [comment]
Thanks for finding that and for the guidance!

---

_The rest of this comment may be more useful to anyone who stumbles across this in the future._

Those references to `@angular-devkit/*` were added when I updated `@ngrx/schematics` from version 5 to 6. It seems that `@angular-devkit/core` and `@angular-devkit/schematics` were peer dependencies for `@ngrx/schematics` back then (in June 2018) and somehow got added to the `devDependencies`. The commits were squashed and I don't recall if I added them to `devDependencies` to address a peer dependency warning. Fortunately those peer dependencies have since been removed, so I expect I'll be able to make the changes you suggested without breaking `@ngrx/schematics` 🙂
